### PR TITLE
test: kafka/handlers admin kafkas and cloud providers unit tests

### PR DESCRIPTION
**Pull request by pawelpaszki**
## Description
test: kafka/handlers admin kafkas and cloud providers unit tests

## Checklist (Definition of Done)

- [x] All acceptance criteria specified in JIRA have been completed
- [x] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- ~~[ ] Documentation added for the feature~~
- [x] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- ~~[ ] Required metrics/dashboards/alerts have been added (or PR created).~~
- ~~[ ] Required Standard Operating Procedure (SOP) is added.~~
- ~~[ ] JIRA has been created for changes required on the client side~~

### CODE REVIEW

File: internal/kafka/internal/handlers/admin_kafka_test.go [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			h := NewAdminKafkaHandler(tt.fields.kafkaService, tt.fields.accountService, tt.fields.providerConfig)
+			req, rw := GetHandlerParams("GET", "/{id}", nil)
+			h.Get(rw, req)
+			Expect(rw.Result().StatusCode).To(Equal(tt.wantStatusCode))
```
Comment (ziccardi) [inline internal/kafka/internal/handlers/admin_kafka_test.go:74]: `rw.Result().Body` must be closed

File: internal/kafka/internal/handlers/admin_kafka_test.go [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			h := NewAdminKafkaHandler(tt.fields.kafkaService, tt.fields.accountService, tt.fields.providerConfig)
+			req, rw := GetHandlerParams("GET", tt.args.url, nil)
+			h.List(rw, req)
+			Expect(rw.Result().StatusCode).To(Equal(tt.wantStatusCode))
```
Comment (ziccardi) [inline internal/kafka/internal/handlers/admin_kafka_test.go:160]: `rw.Result().Body` must be closed

File: internal/kafka/internal/handlers/admin_kafka_test.go [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			h := NewAdminKafkaHandler(tt.fields.kafkaService, tt.fields.accountService, tt.fields.providerConfig)
+			req, rw := GetHandlerParams("DELETE", tt.args.url, nil)
+			h.Delete(rw, req)
+			Expect(rw.Result().StatusCode).To(Equal(tt.wantStatusCode))
```
Comment (ziccardi) [inline internal/kafka/internal/handlers/admin_kafka_test.go:212]: `rw.Result().Body` must be closed

File: internal/kafka/internal/handlers/admin_kafka_test.go [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+			},
+			wantStatusCode: http.StatusBadRequest,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			h := NewAdminKafkaHandler(tt.fields.kafkaService, tt.fields.accountService, tt.fields.providerConfig)
```
Comment (ziccardi) [inline internal/kafka/internal/handlers/admin_kafka_test.go:209]: `tt` is a range variable. It is reused on each iteration. If we run the tests in parallel the value of `tt` will be unpredictable. Better to copy `tt` and use the copy from inside the function.

Comment (pawelpaszki) [reply internal/kafka/internal/handlers/admin_kafka_test.go:209]: does it apply to all test cases? btw. I don't think we are running the tests in parallel

Comment (pawelpaszki) [reply internal/kafka/internal/handlers/admin_kafka_test.go:209]: do you think this should be a separate Jira to do it, including adding documentation?

Comment (ziccardi) [reply internal/kafka/internal/handlers/admin_kafka_test.go:209]: Yeah, I think so. 
I added a point to the tech meeting about adding some linters that will check for these errors.
When we will add the linter, we will have to fix all of these issues as well.

Comment (pawelpaszki) [reply internal/kafka/internal/handlers/admin_kafka_test.go:209]: ok. thanks. I will push the changes for the other comments shortly

File: internal/kafka/internal/handlers/admin_kafka.go [hunk 5]
```diff
@@ -20,14 +20,14 @@ import (
...
 
 type adminKafkaHandler struct {
-	service        services.KafkaService
+	kafkaService   services.KafkaService
 	accountService account.AccountService
 	providerConfig *config.ProviderConfig
 }
 
-func NewAdminKafkaHandler(service services.KafkaService, accountService account.AccountService, providerConfig *config.ProviderConfig) *adminKafkaHandler {
+func NewAdminKafkaHandler(kafkaService services.KafkaService, accountService account.AccountService, providerConfig *config.ProviderConfig) *adminKafkaHandler {
```
Comment (ziccardi) [inline internal/kafka/internal/handlers/admin_kafka.go:28]: This is not related to this PR, but why an exported function (NewAdminKafkaHandler) is returning a non exported type (adminKafkaHandler)?

Comment (pawelpaszki) [reply internal/kafka/internal/handlers/admin_kafka.go:28]: not sure. I can change it if you think it should be exported

Comment (ziccardi) [reply internal/kafka/internal/handlers/admin_kafka.go:28]: I don't think it makes sense that an exported function returns a non exported value: how can code calling this function from outside the package deal with the return value?

I believe visibility should be coherent between functions and return values.

@machi1990 @pawelpaszki  WDYT?

Comment (pawelpaszki) [reply internal/kafka/internal/handlers/admin_kafka.go:28]: yes makes sense to me. @machi1990 - if you agree, I will make this change

Comment (machi1990) [reply internal/kafka/internal/handlers/admin_kafka.go:28]: I think @ziccardi argument makes sense. But I'll defer from doing the change in this PR and create a follow issue to address this, plus do the same for the other handlers as well if it is needed: so far, the handlers have not been really exported, and the route_loader.go (the only place where the handlers are being used) is able to interact with the handlers without any issue.

**Pull request merged**